mangaki/mangaki/management/commands/retrieveposters.py

```diff
@@ -0,0 +1,57 @@
+from django.core.management.base import BaseCommand
+from mangaki.models import Work
+import requests
+
+class Command(BaseCommand):
+    help = 'Downloads posters'
+    def add_arguments(self, parser):
+        parser.add_argument('work_id', nargs='*', type=int)
+        parser.add_argument('--check-exists', action='store_true')
+        parser.add_argument('--ratelimit', type=int, default=10)
+
+    def handle(self, *args, **options):
+        qs = Work.objects.exclude(ext_poster='')
+        if options['work_id']:
+            qs = qs.filter(pk__in=options['work_id'])
+
+        nb_success = 0
+        failed = []
+        recent = []
+        with requests.Session() as s:  # We use a session to use connection pooling
+            num_remaining = len(qs)
+            for work in qs:
+                if not (num_remaining % 10):
+                    self.stdout.write('Remaining: {:d}'.format(num_remaining))
+                num_remaining -= 1
+                if work.int_poster:
+                    if not options['check_exists']:
+                        continue
+                    try:
+                        with work.int_poster.open() as f:
+                            pass
+                    except FileNotFoundError:
+                        pass
+                    else:
+                        continue
+
+                while len(recent) >= options['ratelimit']:
```

Comment:
This condition is never reached, is it?

With the default arguments, `ratelimit = 10` and `len(recent) = 0`.
Could you also add some comments to the selection logic of the recent items? (especially, the `min(max(…))`).